### PR TITLE
Overhauled `RunLoadCounter`

### PR DESCRIPTION
**Pull request by jglick**
* Availability of `TransientActionFactory` obviates the need for `prepare`.
* Current project/count fields were `ThreadLocal` which made it impossible to test the effects of `Queue.maintain`.
* Negative `currCount` logic was hard to follow and blocked clearer logging.

### CODE REVIEW

**Review by Vlatombe — APPROVED** [review]
LGTM

**Comment by basil** [comment]
Causes [a test failure in `PreserveStashesJobPropertyLoadTest#stashWithLoadCount`](https://github.com/jenkinsci/pipeline-model-definition-plugin/pull/623/checks?check_run_id=14106514458) as seen in https://github.com/jenkinsci/pipeline-model-definition-plugin/pull/623.
